src/setup-beam.js

```diff
@@ -175,34 +189,61 @@ async function getRebar3Version(r3Spec) {
 }
 
 async function getOTPVersions(osVersion) {
-  const otpVersionsListing = await get(
-    `https://repo.hex.pm/builds/otp/${osVersion}/builds.txt`,
-  )
-  const otpVersions = new Map()
+  let originListing
+  let pageIdxs
+  if (process.platform === 'linux') {
+    originListing = `https://repo.hex.pm/builds/otp/${osVersion}/builds.txt`
+    pageIdxs = [null]
+  } else if (process.platform === 'win32') {
+    originListing =
+      'https://api.github.com/repos/erlang/otp/releases?per_page=100'
+    pageIdxs = [1, 2, 3]
+  }
 
-  otpVersionsListing
-    .trim()
-    .split('\n')
-    .forEach((line) => {
-      const otpMatch = line.match(/^(OTP-)?([^ ]+)/)
+  const otpVersionsListings = await get(originListing, pageIdxs)
+  const otpVersions = new Map()
 
-      let otpVersion = otpMatch[2]
-      if (semver.validRange(otpVersion)) {
-        otpVersion = semver.minVersion(otpVersion).version
-      }
-      otpVersions.set(otpVersion, otpMatch[0]) // we keep the original for later reference
+  if (process.platform === 'linux') {
+    otpVersionsListings
+      .trim()
+      .split('\n')
+      .forEach((line) => {
+        const otpMatch = line.match(/^(OTP-)?([^ ]+)/)
+
+        let otpVersion = otpMatch[2]
+        if (semver.validRange(otpVersion) && hasPatch(otpVersion)) {
```

Comment:
I added `hasPatch` so that eg. `23.2` doesn't get transformed into `23.2.0`, which doesn't exist.